File: java/3list/src/com/stanislaw/quadrangles/Rectangle.java
package com.stanislaw.quadrangles;

/*
@author: Stanisław Woźniak
@version: 1.0
*/
public class Rectangle extends Quadrangle {

    public Rectangle(double side1, double side3) {
        super(side1, side1, side3, side3, 90);
    }

    @Override
    public double field() {
        return this.side1 * this.side3;
    }

    @Override
    public double perimeter() {
        return this.side1 * 2 + this.side3 * 2;
    }

}

File: C++/3list/Rectangle.cpp
/*
@author: Stanisław Woźniak
@title: rectangle.cpp
*/
#include "Rectangle.hpp"

Rectangle::Rectangle(double side1, double side3){
  this -> side1 = side1;
  side2 = side1;
  this -> side3 = side3;
  side4 = side3;
  angle = 90;
}

double Rectangle::field(){
  return side1*side3;
}

double Rectangle::perimeter(){
  return 4*side1;
}
